compiler/rustc_resolve/src/imports.rs

```diff
@@ -547,6 +547,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             self.import_dummy_binding(*import, is_indeterminate);
 
             if let Some(err) = unresolved_import_error {
+                if let ImportKind::Glob { .. } = import.kind {
+                    self.glob_error = true;
+                }
```

Comment:
@petrochenkov I've tried injecting a nameless binding as a sentinel with things like the following here:

```rust
self.per_ns(|this, ns| {
    this.define(import.parent_scope.module, Ident::with_dummy_span(kw::Empty), ns, this.dummy_binding);
});
```

but then I couldn't find those bindings back in `smart_resolve_path_fragment`. I have access to the module and turn `glob_error` into a `FxHashSet<DefId>` that I check later before emitting the subsequent resolve errors, but that sounds like a bigger hack, inaccurate at that, effectively approximating normal resolve, so I'd like to figure out a way of injecting an unnameable binding as a sentinel so that the regular machinery does its thing, and the error silencing is always accurate (a bad `use foo::*;` won't affect an unrelated scope).